crates/roc_std/src/roc_list.rs

```diff
@@ -586,6 +613,34 @@ where
     }
 }
 
+// This is a RocList that is checked to ensure it is unique or readonly such that it can be sent between threads safely.
+#[repr(transparent)]
+pub struct SendSafeRocList<T>(RocList<T>);
+
+unsafe impl<T> Send for SendSafeRocList<T> where T: Send {}
+
+impl<T> From<RocList<T>> for SendSafeRocList<T>
+where
+    T: Clone,
+{
+    fn from(l: RocList<T>) -> Self {
+        if l.is_unique() || l.is_readonly() {
```

Comment:
does this also need to check if each item is unique/readonly?

Reply:
No, but maybe a better api could be made. Currently, a `RocList` will only be `Send` if every element it contains is `Send`. That comes from the `where T: Send` a few lines up.

As such, If you make a `SendSafeRocList<RocStr>`, it will not be `Send`. You would need to first run a map over the list to create a `SendSafeRocList<SendSafeRocStr>` and then it would be sendable. So this only needs to make the outer list `Send`. Rust will enforce that the inner item is also `Send` before allow it to be sent.

Reply:
that seems reasonable for now. It seems like an OK to manage this automatically would be to define some `ConvertToSendable` trait. I don't know if that's what we want, though!